### PR TITLE
Update for Fastify v5

### PR DESCRIPTION
**Pull request by synapse**
Ref: https://github.com/fastify/fastify/issues/5116

⚠️ There is one small issue with this one, and it's the NextJS dependency. Upgrading it to v14.x will ignore the `basePath` option. Even thought the `basePath` is still present in their config docs and typings it does not work. I am unsure if this [doc](https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration) saying that it was removed might have something to do with this issue. 
I am open to discuss what would be the best way to fix this one.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

CC @simoneb

### CODE REVIEW

**Comment by simoneb** [comment]
created and rebased

**Comment by simoneb** [comment]
tests are broken

**Comment by synapse** [comment]
> tests are broken

I think we need to drop node 14 and 16 for this one

**Comment by simoneb** [comment]
> > tests are broken
> 
> I think we need to drop node 14 and 16 for this one

it was always part of the original plan, wasn't it? https://github.com/fastify/fastify/issues/5116

> Most likely this will simply mean updating dependencies, dropping Node <= 16, and updating CI configuration accordingly

File: test/index.test.js [hunk 1]
```diff
@@ -241,8 +241,8 @@ test('should handle Next initialization errors', async t => {
       prepare: () => Promise.reject(error)
     })
   })
-
-  t.rejects(() => Fastify().register(plugin), error)
+  const fastify = Fastify().register(plugin)
```
Comment (simoneb) [inline test/index.test.js:244]: why would this behavior change? I guess the test was checking that registering the plugin rejected, not awaiting the start of the server

**Review by simoneb — APPROVED** [review]

File: .taprc [hunk 2]
```diff
@@ -1,6 +1,4 @@
-100: true
```
Comment (gurgunday) [inline .taprc:1]: Just one question:

Why did we go to `disable-coverage` from `100`?

Maybe it works without `disable-coverage`?

**Comment by gurgunday** [comment]
Hey, can you pull form main? I can't realistically resolve the package-lock conflicts on the web interface...

And if you could also add node 22 to the test matrix it would be great

Then this is good to go!

**Comment by arthurfiorette** [comment]
@mcollina @gurgunday can we have this release? this package has not yet been released for fastify v5...

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**